Rithika taking over for Vamshi - Update criteria for Bios ready to post to be based on Total Valid Weekly Summaries

guthaVamshi opened the pull request:
# Description
![image](https://github.com/user-attachments/assets/1b67e989-0b52-40ad-8536-4f0e33b0d96a)

### 🛠️ What This PR Does

This update modifies the **filter logic for the Bio Status** in the **Weekly Summaries Report** under:

> `Owner Login → Reports → Weekly Summaries Report`

Previously, qualification was based on the number of weeks a person had been on the team. This has now been corrected to reflect actual participation and contribution.

---

### ✅ Changes Made

- Updated the Bio Status filter criteria to:
  - Require **80 or more total volunteer hours** (already implemented)
  - Require **8 or more valid Weekly Summaries** submitted (instead of 8+ weeks on the team)

---

### 🤔 Why This Matters

This ensures Bio Status is granted only to individuals who have **consistently met expectations**, regardless of how long they’ve been part of the project. It fairly excludes time off, missed weeks, or inactive periods and makes the system **performance-based**.

---

### 🎯 Outcome

Only individuals who have:
- Logged **80+ total volunteer hours**, **AND**
- Submitted **8+ valid Weekly Summaries**

will meet the **Bio Status** criteria.

---

### 🔗 Related PRs

Not related to any other PR.

---

### 🧪 How to Test

1. Check out the current branch
2. Run `npm install` and start the server locally
3. Clear site data/cache
4. Log in as an **Owner** user
5. Navigate to: `Dashboard → Reports → Weekly Summaries Report`
6. Click on **Filter by Bio Status**
7. Verify that results include only users who submitted **8+ valid Weekly Summaries**

---

### 📸 Screenshots / Videos

![image](https://github.com/user-attachments/assets/005f5706-387b-4409-81bc-01a656474e37)


## Note:
Please refresh your cache before testing. Let us know if any edge cases are missed.

nikhilpittala16 reviewed (approved):
I have tested the PR and everything works as expected.

https://github.com/user-attachments/assets/98dbcb56-236f-4edc-a429-6da730e04e87

CasstielP commented:
Tested PR3529, setup was successfully, was able to render the targeted components, UI seemed well proportioned. Bio status toggle buttoned work properly that only displayed with users who submitted 8+ valid weekly summaries. 
![image](https://github.com/user-attachments/assets/d9c2b752-a600-4898-a3c9-d985d0576f92)

guthaVamshi commented:
> Tested PR3529, setup was successfully, was able to render the targeted components, UI seemed well proportioned. Bio status toggle buttoned work properly that only displayed with users who submitted 8+ valid weekly summaries. ![image](https://private-user-images.githubusercontent.com/106496513/447091961-d9c2b752-a600-4898-a3c9-d985d0576f92.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDgyMjU2NjMsIm5iZiI6MTc0ODIyNTM2MywicGF0aCI6Ii8xMDY0OTY1MTMvNDQ3MDkxOTYxLWQ5YzJiNzUyLWE2MDAtNDg5OC1hM2M5LWQ5ODVkMDU3NmY5Mi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNTI2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDUyNlQwMjA5MjNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iNmYyYTI5ZTA3NDNkNmE2MGIxNWNiOGE2YzE3MmViN2U5ZTU0NWI5YjM0ZDFkNTE1YTIyNDAyNDRiOGJiMDViJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.Oi_QqVpUsjs-nyg3hVcrNwE64s8npIBvfOvc9lqJ1Ks)

Hello CasstieIP,

Thank you for reviewing the pull request. Since you seem to be satisfied with the testing, could you please update the status to "Approved" at your convenience?

Appreciate your time and support.

Ram-blip reviewed (dismissed):
Tested the PR and everything looks good! The Bio Status filter now correctly checks for 80+ hours and 8+ valid summaries in both light and dark mode. Great job and Approving the PR. 

<img width="1469" alt="PR 3529" src="https://github.com/user-attachments/assets/70b8c8c1-a629-443b-af54-5bfffa4832d6" />

1709abhishek commented:
<img width="1870" alt="Screenshot 2025-06-07 at 12 23 27 PM" src="https://github.com/user-attachments/assets/d45803d2-aa31-4d28-ab4c-7493003008ca" />
It works fine. tested on my system. one thing i want to mention is, the UI for dark mode, for example the toggle buttons and very non user intuitive.

guthaVamshi committed:
updated weeklysummarycount

kanishkagarwal6101 committed:
Fix Bio Status filter to use weeklySummariesCount instead of daysInTeam

Anusha-Gali reviewed (changes requested):
Hi Kanishk,

I have tried to review your PR locally but was unable to do so due to node version difference. The application should officially support **Node.js v20.x** but your package.json has the node version **"node": ">=14.0.0 <15"**.
<img width="1271" height="754" alt="Screenshot 2025-12-03 at 8 07 57 PM" src="https://github.com/user-attachments/assets/faf493a8-4906-4f54-9d72-9bc8afaec785" />

kanishkagarwal6101 committed:
fix: convert IssueDashboard.css to module CSS

kanishkagarwal6101 committed:
fix: prettier formatting in IssueDashboard

kanishkagarwal6101 committed:
fix: resolve SonarQube code quality issues

- Remove commented code from App.module.css
- Fix overflow shorthand issue (use overflow-y instead of overflow)
- Add generic font family fallback for FontAwesome
- Remove unused styles import from App.jsx
- Merge duplicate .issuesTableResponsive selector in IssueDashboard.module.css
- Remove unused daysInTeam variable from BioFunction.jsx
- Add PropTypes validation for summary.weeklySummariesCount in BioFunction and FormattedReport

kanishkagarwal6101 commented:
Hi all, Thank you for the feedback, I have made the changes requested and fixed the conflicts and code quality issues, I request you to please test the branch again and let me know if it is satisfying the requirements, your feedback is greatly appreciated.

debadyuti23 reviewed (commented):
Hi @kanishkagarwal6101 I have checked the changes in localhost. I can see some entries with weekly valid summaries with 0 as well. I have shared the video of my findings in the comment. Please let me know if this is expected or I might have missed anything.


https://github.com/user-attachments/assets/3eb19b67-9eb5-4327-b1c9-3eba201591d2

kanishkagarwal6101 committed:
fix: update Bio Status filter to use weeklySummariesCount >= 8 instead of daysInTeam > 60

Anusha-Gali reviewed (changes requested):
Hi Kanishk,

I have re-reviewed your PR locally and even though there are summaries whose count is >8 (have shown in the screenshots below), they do not get filtered when i select the "Filter by Bio Status" toggle and it only shows one data.
<img width="1272" height="739" alt="Screenshot 2025-12-16 at 10 20 17 PM" src="https://github.com/user-attachments/assets/e248c811-45b9-4fa5-9ab7-cfa63fc4563f" />
<img width="1272" height="739" alt="Screenshot 2025-12-16 at 10 20 23 PM" src="https://github.com/user-attachments/assets/b3cb27ed-0099-4231-a0a9-bb785a0411cf" />
<img width="1272" height="739" alt="Screenshot 2025-12-16 at 10 20 38 PM" src="https://github.com/user-attachments/assets/598c8e70-3698-4b3d-bfb9-976787225abe" />
<img width="1272" height="739" alt="Screenshot 2025-12-16 at 10 20 57 PM" src="https://github.com/user-attachments/assets/d5e5002d-675a-48e0-b805-8cad67f1f108" />
<img width="1272" height="739" alt="Screenshot 2025-12-16 at 10 21 15 PM" src="https://github.com/user-attachments/assets/91b92b48-6ebc-40e1-be0e-df4b22a3176c" />

ShreyaMP1999 reviewed (commented):
Tested this PR locally as an Owner user. After enabling Filter by Bio Status, results do not match the expected behavior (users with 8+ valid Weekly Summaries). The filtered list still includes users with “Bio announcement: Not requested/posted” and “Weekly Summary: Not provided!”. Also seeing an API error toast (404 on filter list request), which may be affecting the filter logic. Screenshots attached.

<img width="1470" height="956" alt="Screenshot 2026-01-03 at 3 55 59 PM" src="https://github.com/user-attachments/assets/a154bfef-ed95-4977-a654-5972f3c27aa3" />
<img width="1470" height="956" alt="Screenshot 2026-01-03 at 3 56 29 PM" src="https://github.com/user-attachments/assets/2bb89766-8edd-4546-98ef-9d347f3d30ae" />

Shravan-neelamsetty reviewed (commented):
Hi Kanishk, I tested this PR locally by logging in as Owner and navigating to Dashboard → Reports → Weekly Summaries Report. When I enabled "Filter by Bio Status," the results show only 1 user (Total Team Members: 1), despite there being multiple users in the system who should meet the criteria of 8+ valid weekly summaries. The single user shown has 48 valid summaries which exceeds the requirement, but the filter appears to be excluding other users who should also qualify. This suggests the filter logic is either too restrictive or not working correctly.

<img width="1458" height="809" alt="Screenshot 2026-01-09 at 6 38 39 PM" src="https://github.com/user-attachments/assets/9f9c60f9-962e-4338-a6a3-f429df5009fb" />

Vikas-8055 reviewed (commented):
Hi Kanishk, I tested this PR locally by logging in as Owner and navigating to Dashboard → Reports → Weekly Summaries Report. When I enabled "Filter by Bio Status," the results show only 1 user (Total Team Members: 1), despite there being multiple users in the system who should meet the criteria of 8+ valid weekly summaries AND 80+ volunteer hours. The single user shown has 87 valid summaries which exceeds the requirement, but the filter appears to be excluding other users who should also qualify. This suggests the filter logic is either too restrictive or not working correctly.

<img width="1512" height="812" alt="Screenshot 2026-01-10 at 1 26 25 PM" src="https://github.com/user-attachments/assets/bc34b6d6-c67f-4eb8-894a-41f09a4f7c15" />

Ganesh112001 reviewed (approved):
Hi Kanishk, I tested PR #3529 locally and verified that the Bio Status filter criteria update works correctly. After checking out the Vamshi-WeeklySummariesFilter branch and navigating to the Weekly Summaries Report at Dashboard → Reports → Weekly Summaries Report, I applied the "Filter by Bio Status" and confirmed that it now correctly filters users based on the updated criteria. The filter displays only users who have both 80 or more total volunteer hours AND 8 or more valid Weekly Summaries submitted, replacing the previous incorrect criteria that was based on weeks on the team. I verified that users meeting both conditions appear in the filtered results, while users with only one condition met (such as 80+ hours but fewer than 8 summaries, or 8+ summaries but fewer than 80 hours) are correctly excluded. The filter applies quickly in real-time, the clear filter functionality works properly, and no console errors were observed. This change ensures Bio Status is granted fairly based on actual participation and contribution rather than just time on the team, making the system performance-based as intended. Everything functions correctly and is ready for merge!

<img width="1470" height="826" alt="Screenshot 2026-01-16 at 2 45 57 AM" src="https://github.com/user-attachments/assets/f105bfba-c1b4-4de3-b3ae-3fe694c625fe" />

rohanrastogi311 reviewed (approved):
Hi Kanishk, 

Well done with this implementation. 

<img width="1728" height="1037" alt="PR 3529 Screenshot" src="https://github.com/user-attachments/assets/3f8d11ef-4540-40f4-8762-5b110497eb76" />

Vikas-8055 reviewed (approved):
I tested this PR locally by checking out the `Vamshi-WeeklySummariesFilter` branch, running `npm install`, and clearing site data before testing, with screenshots attached for reference. Logged in as an Owner and navigated to Dashboard → Reports → Weekly Summaries Report, enabled the “Filter by Bio Status,” and verified that the filter correctly returns only users who meet both criteria: 80+ total volunteer hours and 8+ valid Weekly Summaries submitted. Users who logged hours but missed weekly summaries are correctly excluded, confirming that the logic now evaluates actual participation rather than time on the team. The feature works as expected in both light and dark modes, with no console errors observed. Approved and ready to merge. 

<img width="1512" height="812" alt="Image" src="https://github.com/user-attachments/assets/6c7d2a6b-ad80-4e87-a0e3-6957d634660a" />

naznin07 reviewed (approved):
Unit tests for BioFunction criteria and dropdown menu logic.
<img width="1811" height="531" alt="PR #3529" src="https://github.com/user-attachments/assets/59dbb1a2-9d12-44e1-b076-d51262d1389f" />

saisandeepkoritala reviewed (commented):
Hi Kaniskh,

I reviewed your PR and it works well in dark and light mode. But the filter seems to have issue. I have more than one entries which have total valid weekly summaries greater than 8 but they are not shown when filter is turned. I can only see 1 record. Please find the screenshots for refrence.

<img width="1808" height="910" alt="Image" src="https://github.com/user-attachments/assets/ff375b56-6dc0-47c8-8b03-4d14e019e059" />
<img width="1782" height="940" alt="Image" src="https://github.com/user-attachments/assets/2d235dbe-97d6-4c2b-a9be-ebe5be6c4c62" />

SharadhaKasiviswanathan commented:
Tested locally on /weeklysummariesreport against the dev backend.
The new weeklySummariesCount >= 8 rule is taking effect, and the report page loaded correctly in my local test. I possibly think of two follow-ups that can be made before merge:
1. The permission description for highlightEligibleBios is still stale. It still says the bio highlight is based on “80 tangible hours, 60 days on the team...”, but the implemented rule is now based on weeklySummariesCount >= 8. Trying to update the permission text so it matches the actual behavior might be helpful.
2. The tests were not fully updated for this rule change. FormattedReport.test.jsx still uses weeklySummariesCount as a string and only raises a prop-type warning, and BioFunction.test.jsx still passes userId as a number and does not cover the new weeklySummariesCount >= 8 eligibility path. Updating/adding regression tests for the new bio-highlight/filter can also be done.

SharadhaKasiviswanathan commented:
I found another issue while testing the Weekly Summaries Report locally:
When switching to the “Last Week” tab and older tabs, the “Filter by Bio Status” toggle behaves opposite to what the UI suggests: with the toggle off, the list is already filtered, and when the toggle is turned on, it stops behaving like a bio-only filter. This looks like a tab specific state or filter sync bug. I have attached the screenshots below.
Please try checking the bio filter behavior across all week tabs and adding a regression test for:
1. toggle off => no bio-status filtering
2. toggle on => only bio-eligible users are shown
3. same behavior remains correct after switching from “This Week” to “Last Week” and older tabs
<img width="1905" height="877" alt="Screenshot 2026-03-29 at 1 22 24 AM" src="https://github.com/user-attachments/assets/a4f5e56c-ab75-40f7-9a8c-2f23e35f5406" />
<img width="1897" height="874" alt="Screenshot 2026-03-29 at 1 22 56 AM" src="https://github.com/user-attachments/assets/0f87b36b-a03d-4f08-9535-f70c1f837943" />

rajanidi1999 reviewed (approved):
This is a good step towards migrating to CSS modules, but there are a few concerns to address before merging. Converting global styles to modules (e.g., App.css) may unintentionally break existing styling across the app, especially with mixed usage of :global. Also noticed some duplicate className props and mixed usage of old and module-based class names, which could lead to bugs or styling inconsistencies. Recommend cleaning these up and verifying UI impact before final approval.

rajanidi1999 reviewed (approved):
Hi,
I have tested your PR and its implemented successfully.

<img width="1920" height="1080" alt="Image" src="https://github.com/user-attachments/assets/557c24b7-1954-4108-a5b2-153cdf787944" />

rithika-paii reviewed (changes requested):
Hi Vikas,

I reviewed PR #3529 and tested the Weekly Summaries Report filter locally in both light and dark modes. The overall functionality is clear, but I found a couple of issues:

Bio Status filter issue: When applying the filter, only one user is displayed, even though multiple users appear to meet the expected criteria (8+ valid weekly summaries). This suggests the filtering logic may not be correctly including all qualifying users.

<img width="1470" height="795" alt="Image" src="https://github.com/user-attachments/assets/de57f2a5-3dcd-4352-a7fb-8742faa04517" />

Dark mode issue: In dark mode, the text on the page is not visible, indicating that the styling is not properly adapted for dark mode. This affects usability and readability.

<img width="1470" height="795" alt="Image" src="https://github.com/user-attachments/assets/fed63ce8-741d-475a-a1b7-6864c00d787f" />

<img width="1470" height="795" alt="Image" src="https://github.com/user-attachments/assets/6cafeb76-4f89-4e16-82db-dc57c2cb49dc" />

Based on these observations, the filter logic and dark mode styling need further fixes before this PR can be approved.

saurabhdipte reviewed (changes requested):
Hi Vikas,

I tested the changes locally, and the results correctly include only users who have submitted 8 or more valid Weekly Summaries, this is working as expected.

However, the dark mode styling needs improvement, as there are contrast issues across the page that affect readability.

<img width="1910" height="937" alt="Image" src="https://github.com/user-attachments/assets/97bcab59-86c6-49b6-8b69-117e182331e9" />

kunchalasireesha reviewed (changes requested):
Hi Vamshi, it looks like this PR currently has merge conflicts with the base branch. Could you please rebase or merge the latest changes from development into your branch and resolve the conflicts? Once the build passes, I’ll finish the review.

kunchalasireesha reviewed (changes requested):
Hi Vamshi, it looks like this PR currently has merge conflicts with the base branch. Could you please rebase or merge the latest changes from development into your branch and resolve the conflicts? Once the build passes, I’ll finish the review.
